rmgpy/rmg/main.py

```diff
@@ -1701,7 +1701,32 @@ def initialize_reaction_threshold_and_react_flags(self):
                         self.bimolecular_react[:num_restart_spcs, :num_restart_spcs] = False
                         if self.trimolecular:
                             self.trimolecular_react[:num_restart_spcs, :num_restart_spcs, :num_restart_spcs] = False
-
+            
+            if self.filter_reactions and self.init_react_tuples:
```

Comment:
Not sure about the reason why `self.filter_reactions` is needed. Do you think we can move this block to the beginning of the function so that values assigned for `self.xxmolecular_react` will not be overwritten by this block. Since we are not really `initialize_reaction_threshold_and_react_flags`(name of the function) in this block. Do you think, it may be better to have a separate function for this block and call the new function in `initialize()`

Reply:
If we aren't filtering reactions there's no need to run this block because everything is going to be reacted anyway and it much better to not tamper with variables that shouldn't be touched in a non-filter run. I've adjusted the block to process identically to the react/threshold update function that should allow it to operate nicely with the restart feature, the values should be changed by this block. I've also isolated it in a separate function as suggested.